lib/couchbase/config_profiles.rb

```diff
@@ -0,0 +1,57 @@
+#  Copyright 2020-2022 Couchbase, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+module Couchbase
+  module ConfigProfiles
+    class Profiles
+      attr :profiles
+
+      def initialize
+        @profiles = {}
+        register_profile("wan_development", DevelopmentProfile.new)
+      end
+
+      def register_profile(name, profile)
+        @profiles[name] = profile
+      end
+
+      def apply(profile_name, options)
+        if @profiles.key?(profile_name)
+          @profiles[profile_name].apply(options)
+          return
+        end
+        raise ArgumentError, "#{profile_name} is not a registered profile"
```

Comment:
this is kind of precondition, lets move it to the beginning of the method, and in this case it could be single line (and bare `return` for positive branch won't be necessary)

```ruby
raise ArgumentError, "#{profile_name} is not a registered profile" unless @profiles.key?(profile_name)

@profiles[profile_name].apply(options)
```